elyra/pipeline/processor.py

```diff
@@ -86,15 +86,7 @@ async def export(self, pipeline, pipeline_export_format, pipeline_export_path, o
 
 
 class PipelineProcessorResponse(object):
-    def __init__(self, run_url, object_storage_url, object_storage_path):
-        # validate that the response has all required properties
-        if not run_url:
-            raise ValueError("Invalid Processor Response: Missing field 'run_url'.")
-        if not object_storage_url:
-            raise ValueError("Invalid Processor Response: Missing field 'object_storage_url'.")
-        if not object_storage_path:
-            raise ValueError("Invalid Processor Response: Missing field 'object_storage_path'.")
-
+    def __init__(self, run_url='', object_storage_url='', object_storage_path=''):
```

Comment:
I still don't really understand the need to change from `None` to the empty string. Is this only because the client can only handle strings and can't handle `None`?  Don't they still need to check either way (length 0 vs no value)?

Also, it seems the local runtimes always return what used to be reserved for successful submissions - even in error cases.  I think I'd rather see exceptions raised in error cases and a link in the exception message to the temporary "ledger directory".  This way, the UX is the same.

cc: @ajbozarth

Reply:
I thought None -> '' was your suggestion... 
See the issue description for UX details.